File: src/version_js/classes.js
import { uuid } from 'uuidv4';

class Break {
    constructor(id, name, breakNumber, startTime, endTime) {
        this.id = id || this._generateId();
        this.name = name;
        this.breakNumber = breakNumber;
        this.startTime = startTime;
        this.endTime = endTime;
    }

    static _generateId() {
        return uuid.v4();
    }
}

class LessonHour {
    constructor(id, lessonNr, startTime, endTime) {
        this.id = id || this._generateId();
        this.lessonNr = lessonNr;
        this.startTime = startTime;
        this.endTime = endTime;
    }

    static _generateId() {
        return uuid.v4();
    }
}

class Lesson {
    constructor(id, subjectId, teacherId, classroomId, lessonHourId, schoolClassGroupId, schoolClassId) {
        this.id = id || this._generateId();
        this.subjectId = subjectId;
        this.teacherId = teacherId;
        this.classroomId = classroomId;
        this.lessonHourId = lessonHourId;
        this.schoolClassGroupId = schoolClassGroupId;
        this.schoolClassId = schoolClassId;
    }

    static _generateId() {
        return uuid.v4();
    }
}

class Schedule {
    constructor(id) {
        this.id = id || this._generateId();
        this.timetable = [];
    }

    _generateId() {
        return uuid.v4();
    }

    addEvent(lesson) {
        this.timetable.push(lesson);
    }
}

class School {
    constructor(id, name, symbol) {
        this.id = id || this._generateId();
        this.name = name;
        this.symbol = symbol;
        //this.type = type;
        //this.nrOfStudents = nrOfStudents;
        this.classrooms = [];
        this.teachers = [];
        this.subjects = [];
    }

    _generateId() {
        return uuid.v4();
    }

    addClassroom(classroom) {
        this.classrooms.push(classroom);
    }
}

class Teacher {
    constructor(id, symbol, possibleClassrooms) {
        //this.type = type;
        this.id = id || this._generateId();
        this.symbol = symbol;
        this.possibleClassrooms = possibleClassrooms || [];
    }

    _generateId() {
        return uuid.v4();
    }
}

class Classroom {
    constructor(id, symbol) {
        //this.name = name;
        //this.type = type;
        //this.capacity = capacity;
        //this.substitutes = substitutes;
        this.id = id || this._generateId();
        this.symbol = symbol;
    }

    _generateId() {
        return uuid.v4();
    }
}

class SchoolClass {
    constructor(id, name, symbol) {
        //this.type = type;
        //this.schoolClassGroups = [];
        //this.nrOfStudents = nrOfStudents;
        this.id = id || this._generateId();
        this.symbol = symbol;
        this.name = name;
    }

    _generateId() {
        return uuid.v4();
    }
}

class SchoolClassGroup {
    constructor(id, symbol, schoolClassId) {
        //this.name = name;
        //this.type = type;
        //this.nrOfStudents = nrOfStudents;
        this.id = id || this._generateId();
        this.symbol = symbol;
        this.schoolClassId = schoolClassId;
    }

    _generateId() {
        return uuid.v4();
    }
    /*
    isLanguage() {
        isType(this, "Language");
    }

    isPE() {
        isType(this, "Physical Education");
    }

    isGirls() {
        isType(this, "Girls");
    }

    isBoys() {
        isType(this, "Boys");
    }

    isVocational() {
        isType(this, "Vocational");
    }
    */
}

class Subject {
    constructor(id, symbol) {
        //this.name = name;
        //this.type = type;
        //this.occurrenceMethods = occurrenceMethods || {"single": 1, "double": 0, "triple": 0};
        //this.possibleSchoolClasses = possibleSchoolClasses;
        //this.possibleClassGroups = possibleClassGroups;
        //this.possibleTeachers = possibleTeachers || [];
        //this.possibleClassrooms = possibleClassrooms || [];
        this.id = id || this._generateId();
        this.symbol = symbol;
    }

    _generateId() {
        return uuid.v4();
    }
    /*
    hasSingleOccurrence() {
        return this.occurrenceMethods.get("single", 0) > 0;
    }

    hasDoubleOccurrence() {
        return this.occurrenceMethods.get("double", 0) > 0;
    }

    hasTripleOccurrence() {
        return this.occurrenceMethods.get("triple", 0) > 0;
    }

    isHumanistic() {
        isType(this, "Humanistic");
    }

    isScientific() {
        isType(this, "Scientific");
    }

    isArtistic() {
        isType(this, "Artistic");
    }
    */
}

class Profile {
    constructor(type, long, short) {
        this.type = type;
        this.long = long;
        this.short = short;
    }
    getShort(){
        return this.short;
    }
}

class TechnicalProfile extends Profile {
    constructor(long, short) {
        super('t', long, short);
    }
}

class MultitradeProfile extends Profile {
    constructor(long, short) {
        super('z', long, short);
    }
}

class HighSchoolProfile extends Profile {
    constructor(long, short) {
        super('lo', long, short);
    }
}

const classProfiles = {
    ek: new TechnicalProfile('Technik ekonomista', 'ekonomik'),
    fry: new TechnicalProfile('Technik usług fryzjerskich', 'fryzjer'),
    gas: new TechnicalProfile('Technik żywienia i usług gastronomicznych', 'gastronom'),
    ha: new TechnicalProfile('Technik handlowiec', 'handlowiec'),
    hot: new TechnicalProfile('Technik hotelarstwa', 'hotelarz'),
    log: new TechnicalProfile('Technik logistyk', 'logistyk'),
    ra: new TechnicalProfile('Technik rachunkowości', 'rachunkowość'),
    sport: new HighSchoolProfile('Klasa sportowa', 'sportowa'),
    wz: new MultitradeProfile('Oddział wielozawodowy', 'wielozawodowa')
};

export {Profile, TechnicalProfile, MultitradeProfile, classProfiles};

File: src/.old/classes.py
import uuid
from utils import *


class Break:
    def __init__(self, id, name, breakNumber, startTime, endTime):
        self.id = id or self._generateId()
        self.name = name
        self.breakNumber = breakNumber
        self.startTime = startTime
        self.endTime = endTime

    @staticmethod
    def _generateId():
        return str(uuid.uuid4())



class LessonHour:
    def __init__(self, id, lessonNr, startTime, endTime):
        self.id = id or self._generateId()
        self.lessonNr = lessonNr
        self.startTime = startTime
        self.endTime = endTime

    @staticmethod
    def _generateId():
        return str(uuid.uuid4())

class Lesson:
    def __init__(self, id, subjectId, teacherId, classroomId, lessonHourId, schoolClassGroupId, schoolClassId):
        self.id = id or self._generateId()
        self.subjectId = subjectId
        self.teacherId = teacherId
        self.classroomId = classroomId
        self.lessonHourId = lessonHourId
        self.schoolClassGroupId = schoolClassGroupId
        self.schoolClassId = schoolClassId

    @staticmethod
    def _generateId():
        return str(uuid.uuid4())



class Schedule:
    def __init__(self, id):
        self.id = id or self._generateId()
        self.timetable = []

    @staticmethod
    def _generateId():
        return str(uuid.uuid4())

    def addEvent(self, lesson):
        self.timetable.append(lesson)



class School:
    def __init__(self, id, name, symbol):
        #, type, nrOfStudents
        self.id = id or self._generateId()
        self.name = name
        self.symbol = symbol
        '''self.type = type
        self.nrOfStudents = nrOfStudents'''
        self.classrooms = []
        self.teachers = []
        self.subjects = []

    @staticmethod
    def _generateId():
        return str(uuid.uuid4())

    def addClassroom(self, classroom):
        self.classrooms.append(classroom)



class Teacher:
    def __init__(self, id, symbol, possibleClassrooms):
        #, type
        self.id = id or self._generateId()
        self.symbol = symbol
        '''self.type = type'''
        self.possibleClassrooms = possibleClassrooms or []

    @staticmethod
    def _generateId():
        return str(uuid.uuid4())



class Classroom:
    def __init__(self, id, symbol):
        #, name, type, capacity, substitute
        self.id = id or self._generateId()
        self.symbol = symbol
        '''self.name = name
        self.type = type
        self.capacity = capacity
        self.substitutes = substitutes'''

    @staticmethod
    def _generateId():
        return str(uuid.uuid4())



class SchoolClass:
    def __init__(self, id, name, symbol):
        #, type, schoolClassGroups, nrOfStudents
        self.id = id or self._generateId()
        self.symbol = symbol
        self.name = name
        '''self.type = type
        self.schoolClassGroups = []
        self.nrOfStudents = nrOfStudents'''

    @staticmethod
    def _generateId():
        return str(uuid.uuid4())



class SchoolClassGroup:
    def __init__(self, id, symbol, schoolClassId):
        #, name, type, nrOfStudents
        self.id = id if id else self._generateId()
        self.symbol = symbol
        self.schoolClassId = schoolClassId
        '''self.name = name
        self.type = type
        self.nrOfStudents = nrOfStudents'''

    @staticmethod
    def _generateId():
        return str(uuid.uuid4())

    '''def isLanguage(self):
        isType(self, "Language")

    def isPE(self):
        isType(self, "Physical Education")

    def isGirls(self):
        isType(self, "Girls")
    
    def isBoys(self):
        isType(self, "Boys")

    def isVocational(self):
        isType(self, "Vocational")'''



class Subject:
    def __init__(self, id, symbol):
        #, name, type, occurrenceMethods, possibleSchoolClasses, possibleClassGroups, possibleTeachers, possibleClassrooms
        self.id = id if id else self._generateId()
        self.symbol = symbol
        '''self.name = name
        self.type = type
        self.occurrenceMethods = occurrenceMethods or {"single": 1, "double": 0, "triple": 0}
        self.possibleSchoolClasses = possibleSchoolClasses
        self.possibleClassGroups = possibleClassGroups
        self.possibleTeachers = possibleTeachers or []
        self.possibleClassrooms = possibleClassrooms or []'''

    @staticmethod
    def _generateId():
        return str(uuid.uuid4())

    '''def hasSingleOccurrence(self):
        return self.occurrenceMethods.get("single", 0) > 0

    def hasDoubleOccurrence(self):
        return self.occurrenceMethods.get("double", 0) > 0

    def hasTripleOccurrence(self):
        return self.occurrenceMethods.get("triple", 0) > 0

    def isHumanistic(self):
        isType(self, "Humanistic")

    def isScientific(self):
        isType(self, "Scientific")

    def isArtistic(self):
        isType(self, "Artistic")'''



class Profile:
    def __init__(self, type, long, short):
        self.type = type
        self.long = long
        self.short = short

    def getShort(self):
        return self.short

class TechnicalProfile(Profile):
    def __init__(self, long, short):
        super().__init__('t', long, short)

class MultitradeProfile(Profile):
    def __init__(self, long, short):
        super().__init__('z', long, short)

class HighSchoolProfile(Profile):
    def __init__(self, long, short):
        super().__init__('lo', long, short)


classProfiles = {
    'ek': TechnicalProfile('Technik ekonomista', 'ekonomik'),
    'fry': TechnicalProfile('Technik usług fryzjerskich', 'fryzjer'),
    'gas': TechnicalProfile('Technik żywienia i usług gastronomicznych', 'gastronom'),
    'ha': TechnicalProfile('Technik handlowiec', 'handlowiec'),
    'hot': TechnicalProfile('Technik hotelarstwa', 'hotelarz'),
    'log': TechnicalProfile('Technik logistyk', 'logistyk'),
    'ra': TechnicalProfile('Technik rachunkowości', 'rachunkowość'),
    'sport': HighSchoolProfile('Klasa sportowa', 'sportowa'),
    'wz': MultitradeProfile('Oddział wielozawodowy', 'wielozawodowa')
}